renderer.php

```diff
@@ -274,58 +299,93 @@ public function presignedurl_tests_load_files($fs) {
         }
 
         $output .= $this->box('');
         $output .= $this->heading(get_string('presignedurl_testing:test3', 'tool_objectfs'), 4);
         foreach ($testfiles as $file) {
             $headers = array('Content-Disposition: inline; filename="'.$file->get_filename().
                 '"', 'Content-Type: '.$file->get_mimetype());
-            $presignedurl = $fs->generate_presigned_url_to_external_file($file->get_contenthash(), $headers);
+            $presignedurl = $this->generate_presigned_url($fs, $file, $headers);
 
-            $outputstring = get_string('presignedurl_testing:openinbrowser', 'tool_objectfs').': '.
-                '<a href="'.$presignedurl.'">'.$file->get_filename().'</a>';
-            $output .= $this->heading($outputstring, 5);
+            $output .= $this->heading($this->get_output($presignedurl, $file->get_filename(), 'openinbrowser'), 5);
         }
 
         $output .= $this->box('');
         $output .= $this->heading(get_string('presignedurl_testing:test4', 'tool_objectfs'), 4);
         foreach ($testfiles as $file) {
             $headers = array('Content-Disposition: inline; filename="'.$file->get_filename().
                 '"', 'Content-Type: '.$file->get_mimetype());
-            $presignedurl = $fs->generate_presigned_url_to_external_file($file->get_contenthash(), $headers);
+            $presignedurl = $this->generate_presigned_url($fs, $file, $headers);
 
             $outputstring = '"'.$file->get_filename().'" '.get_string('presignedurl_testing:fileiniframe', 'tool_objectfs').':';
             $output .= $this->heading($outputstring, 5);
 
-            $outputstring = '<iframe height="400" width="100%" src="'.$presignedurl.'">'.
-                get_string('presignedurl_testing:iframesnotsupported', 'tool_objectfs').'</iframe>';
-            $output .= $this->box($outputstring);
+            $output .= $this->box($this->get_output($presignedurl, $file->get_filename(), 'iframesnotsupported'));
             $output .= $this->box('');
         }
 
         return $output;
     }
+
+    /**
+     * Generates the output string that contains the presignedurl.
+     * @param string $url
+     * @param string $filename
+     * @param string $identifier
+     * @return string
+     * @throws coding_exception
+     */
+    private function get_output($url, $filename, $identifier) {
+        if ('iframesnotsupported' === $identifier) {
+            $output = '<iframe height="400" width="100%" src="' . $url . '">'.
+                get_string('presignedurl_testing:' . $identifier, 'tool_objectfs').'</iframe>';
+        } else {
+            $output = get_string('presignedurl_testing:' . $identifier, 'tool_objectfs').': '.
+                '<a href="'. $url .'">'. $filename . '</a>';
+        }
+        return $output . '<br><small>Redirecting to: ' . pathinfo($url, PATHINFO_DIRNAME) . '/...</small>';;
```

Comment:
I like this, can you improve this so that it shows when it will or won't redirect?